test/inductor/test_fused_attention.py

```diff
@@ -311,10 +333,10 @@ def forward(self, query, key, value, scale_factor) -> torch.Tensor:
         tensor_shape = (2, 4, 4, 4)
         for is_inv_factor in [True, False]:
             args = [
-                torch.randn(tensor_shape, device="cuda"),
-                torch.randn(tensor_shape, device="cuda"),
-                torch.randn(tensor_shape, device="cuda"),
-                torch.randn((4, 1, 1), device="cuda"),
+                torch.randn(tensor_shape),
```

Comment:
are we still testing cuda at all ?

Reply:
Yes, both CUDA and CPU paths are under testing now. I remove the restrictions of `HAS_CUDA` and `PLATFORM_SUPPORTS_FUSED_SDPA` when entering the `run_tests()`.

Reply:
To be more clear, I refactored the UT by creating a base class `TestSDPAPatternRewriterTemplate` and two sub-classes respectively for CUDA and CPU.